src/pages/NodesTelemetry.vue

```diff
@@ -0,0 +1,397 @@
+<script setup lang="ts">
+import * as http from '@/shared/api';
+import BaseTable from '@/shared/ui/components/BaseTable.vue';
+import BaseContentBlock from '@/shared/ui/components/BaseContentBlock.vue';
+import { computed, reactive, watch } from 'vue';
+import { formatNumber } from '@/shared/ui/utils/formatters';
+import type { PeerInfo } from '@/shared/api/schemas';
+import { PeerMetrics } from '@/shared/api/schemas';
+import { useErrorHandlers } from '@/shared/ui/composables/useErrorHandlers';
+import BaseLoading from '@/shared/ui/components/BaseLoading.vue';
+import { useIntervalFn } from '@vueuse/shared';
+import { useAsyncState, useWindowSize } from '@vueuse/core';
+import BaseHash from '@/shared/ui/components/BaseHash.vue';
+import { LG_WINDOW_SIZE, MD_WINDOW_SIZE, SM_WINDOW_SIZE, XS_WINDOW_SIZE } from '@/shared/ui/consts';
+import LatestBlock from '@/entities/telemetry/LatestBlock.vue';
+import { streamPeerMetrics } from '@/shared/api';
+
+const { handleUnknownError } = useErrorHandlers();
+
+const { width } = useWindowSize();
+
+const hashType = computed(() => {
+  if (width.value >= LG_WINDOW_SIZE) return 'medium';
+
+  if (width.value >= MD_WINDOW_SIZE) return 'short';
+
+  if (width.value >= SM_WINDOW_SIZE) return 'medium';
+
+  if (width.value >= XS_WINDOW_SIZE) return 'short';
+
+  return 'short';
+});
+
+interface PeerData {
+  metrics: null | PeerMetrics
+  info: null | PeerInfo
+}
+
+const peers = reactive(new Map<string, PeerData>());
+
+const {
+  isLoading: isMetricsLoading,
+  state: metrics,
+  execute: getNetworkMetrics,
+} = useAsyncState(http.fetchNetworkMetrics, null, {
+  immediate: false,
+  onError: handleUnknownError,
+  resetOnExecute: false,
+});
+const { state: peersInfo } = useAsyncState(http.fetchPeersInfo, null, {
+  immediate: true,
+  onError: handleUnknownError,
+  onSuccess: (data) => {
+    data?.forEach((i) => {
+      peers.set(i.public_key, { info: { ...i }, metrics: null });
+    });
+  },
+});
+
+const { data: streamedPeerMetrics, status: streamStatus } = streamPeerMetrics();
+
+watch(streamedPeerMetrics, () => {
+  if (!streamedPeerMetrics.value || !peersInfo.value) return;
```

Comment:
This is not correct.  

```suggestion
  if (!streamedPeerMetrics.value) return;
```

Metrics could arrive earlier. On the first connection, they arrive for all peers. What happens for me locally is that data is completely disregarded until `info` is loaded.

Reply:
Removed✅